gating/check/pre_deploy.sh

```diff
@@ -60,8 +60,21 @@ if [[ ! -e phantomjs-2.1.1-linux-x86_64.tar.bz2 ]]; then
   tar -xjf phantomjs-2.1.1-linux-x86_64.tar.bz2
 fi
 
+# FrankZhang: Limit venv python to python2 
+# to avoid python3 code making syntax error https://rpc-openstack.atlassian.net/browse/RI-458
 if [[ ! -d ".venv" ]]; then
-  virtualenv .venv
+  virtualenv .venv --python=python2 --always-copy --no-download
+fi
+
+# FrankZhang: Upgrade pip and setuptools due to outdate issue https://rpc-openstack.atlassian.net/browse/RI-458
+pip_version=$(pip -V | awk '{print $2}')
+if [[ $pip_version < 9.0.1 ]]; then
+    pip install --upgrade pip==9.0.1
```

Comment:
Why are we picking this version?  If we hardcode we'll have issues in the future. (same for setuptools below).

    pip install -c{env:UPPER_CONSTRAINTS_FILE:https://git.openstack.org/cgit/openstack/openstack-ansible/plain/global-requirement-pins.txt?h=stable/rocky} --upgrade pip

same for setuptools below (with the branch name being a variable preferably).

Reply:
actually, this whole pip section could be one line.

    pip install -c{env:UPPER_CONSTRAINTS_FILE:https://git.openstack.org/cgit/openstack/openstack-ansible/plain/global-requirement-pins.txt?h=stable/rocky} --upgrade pip setuptools

again, with the branch name being a variable preferably

Reply:
I would suggest using the newton constraints from OSA/openstack given that this branch is newton.

Reply:
ya, newton would be best in this case, but hardcoding should be avoided in general if this is going into any other branch.

Reply:
as far as upgrades go, the command I gave will ALWAYS install the version defined in the link given.

Reply:
so would this one be what we want to use? https://git.openstack.org/cgit/openstack/openstack-ansible/tree/global-requirement-pins.txt?h=newton-eol

Reply:
yep

Reply:
Actually, it'd be https://raw.githubusercontent.com/rcbops/openstack-ansible/stable/newton/global-requirement-pins.txt given that we have our own fork of OSA's newton.